flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/InputChannel.java

```diff
@@ -305,19 +311,18 @@ protected CheckpointBarrier parseCheckpointBarrierOrNull(Buffer buffer) throws I
 
 	/**
 	 * A combination of a {@link Buffer} and a flag indicating availability of further buffers,
-	 * and the backlog length indicating how many non-event buffers are available in the
-	 * subpartition.
+	 * and the backlog length indicating how many credits the subpartition.
 	 */
 	public static final class BufferAndAvailability {
 
 		private final Buffer buffer;
 		private final boolean moreAvailable;
-		private final int buffersInBacklog;
+		private final int backlog;
 
-		public BufferAndAvailability(Buffer buffer, boolean moreAvailable, int buffersInBacklog) {
-			this.buffer = checkNotNull(buffer);
+		public BufferAndAvailability(@Nullable Buffer buffer, boolean moreAvailable, int backlog) {
```

Comment:
We can also avoid introducing `nullable` buffer by this comment https://github.com/apache/flink/pull/11877/files#r422847423